### PR TITLE
RD-3222 Improve relationships between ESLint configs

### PR DESCRIPTION
**Pull request by qooban**
## Description
Changed ESLint common configurations to allow easier overriding:
1. Make `eslint-common` as a base for all, with an intention not to use it directly (with only one exception: `cloudify-blueprint-topology`)
2. Make all ESLint configurations to base either on `airbnb` (React) or `airbnb-base` (non-React)
3. Fix minor issues inside configs (see comments)

The following changes are necessary in the other repositories:
1. https://github.com/cloudify-cosmo/cloudify-ui-components/pull/123 
2. https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/188
3. https://github.com/cloudify-cosmo/cloudify-blueprint-composer/pull/960
4. https://github.com/cloudify-cosmo/cloudify-stage/pull/1796 

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
CI pipeline is green.

In all the branches created for the following PRs: 
    1. https://github.com/cloudify-cosmo/cloudify-ui-components/pull/123 
    2. https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/188
    3. https://github.com/cloudify-cosmo/cloudify-blueprint-composer/pull/960
    4. https://github.com/cloudify-cosmo/cloudify-stage/pull/1796 

did the following
```
npm link ../cloudify-ui-common
npm run lint
```
In all cases it was passing.

## Documentation
Updated `configs/README.md`.

### CODE REVIEW

File: backend/.eslintrc.json [hunk 1]
```diff
@@ -2,8 +2,6 @@
     "extends": ["../configs/eslint-common-node.json"],
     "rules": {
         // NOTE: packageDir is relative to the current working directory when executing linter
-        "import/no-extraneous-dependencies": ["error", { "packageDir": "./" }],
-        // NOTE: handled by import/no-extraneous-dependencies
```
Comment (qooban) [inline backend/.eslintrc.json:6]: It's now part of `eslint-common-node.json`.

File: configs/eslint-common-cypress.json [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
-
+    "extends": [
+        "airbnb-base",
+        "./eslint-common.json",
+        "plugin:chai-friendly/recommended",
+        "plugin:cypress/recommended"
+    ],
+    "rules": {
+        "import/no-extraneous-dependencies": "off"
+    },
```
Comment (qooban) [inline configs/eslint-common-cypress.json:10]: It make sense to add it here, as we did that in local configs for Cypress in Stage / Composer.

File: configs/eslint-common-node.json [hunk 3]
```diff
@@ -1,5 +1,19 @@
...
+    "extends": [
+        "airbnb-base",
+        "./eslint-common.json",
+        "plugin:node/recommended"
+    ],
+    "rules": {
+        "node/no-unsupported-features/es-syntax": "off",
+        // NOTE: handled by import/no-extraneous-dependencies
+        "node/no-extraneous-require": "off"
+    },
```
Comment (qooban) [inline configs/eslint-common-node.json:11]: There are probably more rules which can be disabled as we are moving towards having fully TS codebase, but I think it's not for this PR.

File: configs/eslint-common-react.json [hunk 4]
```diff
@@ -1,10 +1,8 @@
 {
     "extends": [
-        "./eslint-common.json",
         "airbnb",
```
Comment (qooban) [inline configs/eslint-common-react.json:3]: That was the main problem encountered by @leon-good-life with his PR - https://github.com/cloudify-cosmo/cloudify-stage/pull/1796 . Order matters in ESLint, and `airbnb` was overwriting changes introduced to `eslint-common.json`.

File: configs/eslint-common-react.json [hunk 5]
```diff
@@ -1,10 +1,8 @@
...
         "prettier/react",
-        "plugin:prettier/recommended",
-        "./eslint-ts-overrides.json"
+        "./eslint-common.json"
```
Comment (qooban) [inline configs/eslint-common-react.json:5]: Both `plugin:prettier/recommended` and `eslint-ts-overrides.json` are part of `eslint-common.json`, so it was redundant here.

File: configs/eslint-common.json [hunk 6]
```diff
@@ -58,7 +57,7 @@
     },
     "overrides": [
         {
-            "files": ["*.ts"],
+            "files": ["*.ts", "*.tsx"],
```
Comment (kubama) [inline configs/eslint-common.json:60]: Why this change?

Comment (qooban) [reply configs/eslint-common.json]: Right, good catch. We agreed to turn it off only for TS files (RD-2112) my mistake. Reverted that change and updated https://github.com/cloudify-cosmo/cloudify-stage/pull/1796 accordingly. Other PRs not affected.

**Review by kubama — APPROVED** [review]
